lib/database.js

```diff
@@ -150,14 +150,15 @@ module.exports = class Database extends EventEmitter {
 
         return this.connection;
       });
+    } else {
+      return Promise.reject(new Error(`Connection string of type ${typeof this.connectionString} not supported.`));
```

Comment:
just a small tweak since it can also be another mongoist object
```suggestion
      return Promise.reject(new Error(`Connection parameter of type ${typeof this.connectionString} not supported.`));
```